src/libexpr/flake/flake.cc

```diff
@@ -298,6 +298,9 @@ LockedFlake lockFlake(
 
     auto flake = getFlake(state, topRef, lockFlags.useRegistries, flakeCache);
 
+    flake.config.apply();
```

Comment:
Unless I miss something, this will also apply for the `getFlake` builtin, which would be quite impure.

Maybe you can add a new field to `LockFlags` so that it’s only triggerd conditionally

Reply:
Good point!

Added a flag and set it true when:

- used through `getLockedFlake` in installable
- `nix flake {update,lock}`